apollo-federation/src/sources/connect/validation/http/body.rs

```diff
@@ -0,0 +1,156 @@
+use std::ops::Range;
+
+use apollo_compiler::ast::Value;
+use apollo_compiler::parser::LineColumn;
+use apollo_compiler::Name;
+use apollo_compiler::Node;
+
+use crate::sources::connect::json_selection::KnownVariable;
+use crate::sources::connect::json_selection::PathList;
+use crate::sources::connect::json_selection::Ranged;
+use crate::sources::connect::json_selection::WithRange;
+use crate::sources::connect::spec::schema::CONNECT_BODY_ARGUMENT_NAME;
+use crate::sources::connect::validation::coordinates::BodyCoordinate;
+use crate::sources::connect::validation::coordinates::ConnectDirectiveCoordinate;
+use crate::sources::connect::validation::graphql::GraphQLString;
+use crate::sources::connect::validation::graphql::SchemaInfo;
+use crate::sources::connect::validation::selection::get_json_selection;
+use crate::sources::connect::validation::Code;
+use crate::sources::connect::validation::Message;
+use crate::sources::connect::JSONSelection;
+use crate::sources::connect::PathSelection;
+
+pub(crate) fn validate_body(
+    http_arg: &[(Name, Node<Value>)],
+    connect_coordinate: ConnectDirectiveCoordinate,
+    schema: &SchemaInfo,
+) -> Result<(), Message> {
+    let Some(body) = http_arg
+        .iter()
+        .find_map(|(name, value)| (name == &CONNECT_BODY_ARGUMENT_NAME).then_some(value))
+    else {
+        return Ok(());
+    };
+    let coordinate = BodyCoordinate::from(connect_coordinate);
+    let (value, json_selection) = get_json_selection(coordinate, &schema.sources, body)?;
+    let arg = Arg {
+        value,
+        coordinate,
+        schema,
+    };
+
+    match json_selection {
+        JSONSelection::Named(_) => Ok(()), // TODO: validate_sub_selection(sub_selection, arg),
+        JSONSelection::Path(path_selection) => validate_path_selection(path_selection, arg),
+    }
+}
+
+fn validate_path_selection(path_selection: PathSelection, arg: Arg) -> Result<(), Message> {
+    let PathSelection { path } = path_selection;
+    match path.as_ref() {
+        PathList::Var(variable, trailing) => {
```

Comment:
Probably need to process `trailing` in more than just the `$args` case, since other `$args` paths could appear elsewhere along the path, within method arguments for example.